vm/src/serde/deserialize_program.rs

```diff
@@ -812,43 +816,60 @@ mod tests {
-                    String::from("starkware.cairo.common.alloc"),
-                    String::from("starkware.cairo.common.alloc.alloc"),
-                ],
-                flow_tracking_data: FlowTrackingData {
-                    ap_tracking: ApTracking {
-                        group: 0,
-                        offset: 0,
+        let hints: HashMap<_, _> = [
+            (
+                0,
+                vec![HintParams {
+                    code: "memory[ap] = segments.add()".to_string(),
+                    accessible_scopes: vec![
+                        String::from("starkware.cairo.common.alloc"),
+                        String::from("starkware.cairo.common.alloc.alloc"),
+                    ],
+                    flow_tracking_data: FlowTrackingData {
+                        ap_tracking: ApTracking {
+                            group: 0,
+                            offset: 0,
+                        },
+                        reference_ids: HashMap::new(),
                     },
-                    reference_ids: HashMap::new(),
-                },
-            }],
-        );
-        hints.insert(
-            46,
-            vec![HintParams {
-                code: "import math".to_string(),
-                accessible_scopes: vec![String::from("__main__"), String::from("__main__.main")],
-                flow_tracking_data: FlowTrackingData {
-                    ap_tracking: ApTracking {
-                        group: 5,
-                        offset: 0,
+                }],
+            ),
+            (
+                46,
+                vec![HintParams {
+                    code: "import math".to_string(),
+                    accessible_scopes: vec![
+                        String::from("__main__"),
+                        String::from("__main__.main"),
+                    ],
+                    flow_tracking_data: FlowTrackingData {
+                        ap_tracking: ApTracking {
+                            group: 5,
+                            offset: 0,
+                        },
+                        reference_ids: HashMap::new(),
                     },
-                    reference_ids: HashMap::new(),
-                },
-            }],
-        );
+                }],
+            ),
+        ]
+        .into();
+        let mut hints_ranges = vec![None; 47];
+        hints_ranges[0] = Some((0, NonZeroUsize::new(1).unwrap()));
+        hints_ranges[46] = Some((1, NonZeroUsize::new(1).unwrap()));
 
         assert_eq!(program.builtins, builtins);
         assert_eq!(program.shared_program_data.data, data);
         assert_eq!(program.shared_program_data.main, Some(0));
-        assert_eq!(program.shared_program_data.hints, hints);
+
+        let program_hints: HashMap<_, _> = program
+            .shared_program_data
+            .hints_ranges
+            .iter()
+            .enumerate()
+            .filter_map(|(pc, r)| r.map(|(s, l)| (pc, (s, s + l.get()))))
```

Comment:
this is a bit confusing, maybe a comment explaining what is happening would be useful.

Reply:
Thanks, I'll do that, or maybe I'll see if I can make it simpler.